compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -4174,37 +4174,76 @@ object Types {
 
     def tryCompiletimeConstantFold(using Context): Type = tycon match {
       case tycon: TypeRef if defn.isCompiletimeAppliedType(tycon.symbol) =>
-        def constValue(tp: Type): Option[Any] = tp.dealias match {
+        extension (tp : Type) def fixForEvaluation : Type =
+          tp.normalized.dealias match {
+            case tp : TermRef => tp.underlying
+            case tp => tp
+          }
+
+        def constValue(tp: Type): Option[Any] = tp.fixForEvaluation match {
           case ConstantType(Constant(n)) => Some(n)
           case _ => None
         }
 
-        def boolValue(tp: Type): Option[Boolean] = tp.dealias match {
+        def boolValue(tp: Type): Option[Boolean] = tp.fixForEvaluation match {
           case ConstantType(Constant(n: Boolean)) => Some(n)
           case _ => None
         }
 
-        def intValue(tp: Type): Option[Int] = tp.dealias match {
+        def intValue(tp: Type): Option[Int] = tp.fixForEvaluation match {
           case ConstantType(Constant(n: Int)) => Some(n)
           case _ => None
         }
 
-        def stringValue(tp: Type): Option[String] = tp.dealias match {
-          case ConstantType(Constant(n: String)) => Some(n)
+        def longValue(tp: Type): Option[Long] = tp.fixForEvaluation match {
+          case ConstantType(Constant(n: Long)) => Some(n)
+          case _ => None
+        }
+
+        def floatValue(tp: Type): Option[Float] = tp.fixForEvaluation match {
+          case ConstantType(Constant(n: Float)) => Some(n)
+          case _ => None
+        }
+
+        def doubleValue(tp: Type): Option[Double] = tp.fixForEvaluation match {
+          case ConstantType(Constant(n: Double)) => Some(n)
           case _ => None
         }
 
+        def stringValue(tp: Type): Option[String] = tp.fixForEvaluation match {
+          case ConstantType(Constant(n: String)) => Some(n)
+          case _ => None
+        }
+        def isConst : Option[Type] = args.head.fixForEvaluation match {
+          case ConstantType(_) => Some(ConstantType(Constant(true)))
+          case _ => Some(ConstantType(Constant(false)))
+        }
```

Comment:
For consistency with the other methods, I think we should:

- Make `isConst` take a parameter `tp: Type` instead of automatically applying to `args.head`
- (nitpick: skip a line around the function)